tests/components/config/test_entity_registry.py

```diff
@@ -142,6 +142,187 @@ class Unserializable:
     ]
 
 
+async def test_list_entities_for_display(hass, client):
+    """Test list entries."""
+    mock_registry(
+        hass,
+        {
+            "test_domain.test": RegistryEntry(
+                area_id="area52",
+                device_id="device123",
+                entity_id="test_domain.test",
+                has_entity_name=True,
+                original_name="Hello World",
+                platform="test_platform",
+                unique_id="1234",
+            ),
+            "test_domain.nameless": RegistryEntry(
+                area_id="area52",
+                device_id="device123",
+                entity_id="test_domain.nameless",
+                has_entity_name=True,
+                original_name=None,
+                platform="test_platform",
+                unique_id="1234",
+            ),
+            "test_domain.renamed": RegistryEntry(
+                area_id="area52",
+                device_id="device123",
+                entity_id="test_domain.renamed",
+                has_entity_name=True,
+                name="User name",
+                original_name="Hello World",
+                platform="test_platform",
+                unique_id="1234",
+            ),
+            "test_domain.boring": RegistryEntry(
+                entity_id="test_domain.boring",
+                platform="test_platform",
+                unique_id="2345",
+            ),
+            "test_domain.disabled": RegistryEntry(
+                disabled_by=RegistryEntryDisabler.USER,
+                entity_id="test_domain.disabled",
+                hidden_by=RegistryEntryHider.USER,
+                platform="test_platform",
+                unique_id="4567",
+            ),
+            "test_domain.hidden": RegistryEntry(
+                entity_id="test_domain.hidden",
+                hidden_by=RegistryEntryHider.USER,
+                platform="test_platform",
+                unique_id="3456",
+            ),
+            "sensor.default_precision": RegistryEntry(
+                entity_id="sensor.default_precision",
+                options={"sensor": {"suggested_display_precision": 0}},
+                platform="test_platform",
+                unique_id="5678",
+            ),
+            "sensor.user_precision": RegistryEntry(
+                entity_id="sensor.user_precision",
+                options={
+                    "sensor": {"display_precision": 0, "suggested_display_precision": 1}
+                },
+                platform="test_platform",
+                unique_id="6789",
+            ),
+        },
+    )
+
+    await client.send_json({"id": 5, "type": "config/entity_registry/list_for_display"})
+    msg = await client.receive_json()
+
+    assert msg["result"] == {
+        "entity_categories": {"0": "config", "1": "diagnostic"},
```

Comment:
Should we also have area ID lookup and device ID lookup?

Reply:
That'll be a bit more work, because we then need to invalidate cached display dicts whenever something changes.
I'd say we let the zlib compression of WS commands handle deduplication of area and device id for now.